### ISSUE TITLE
Fix lock (key - "kmm-asset-lock") not obtained

### ISSUE THREAD

**Issue by lewismarshall** [UKHomeOffice/keto-k8#9]
After failing to create a control plane, the kmm terminates cleanly (with error) but lock is NOT cleaned up OR can't be re-obtained:
```
Apr 19 16:03:19 ip-172-31-4-32.eu-west-2.compute.internal docker[24422]: time="2017-04-19T16:03:19Z" level=info msg="Saving:"/etc/kubernetes/scheduler.conf""
Apr 19 16:03:19 ip-172-31-4-32.eu-west-2.compute.internal docker[24422]: [apiclient] Created API client, waiting for the control plane to become ready
Job for kmm.service failed because the control process exited with error code.
See "systemctl status kmm.service" and "journalctl -xe" for details.
Apr 19 16:04:37 ip-172-31-4-32.eu-west-2.compute.internal systemd[1]: kmm.service: Start-pre operation timed out. Terminating.
Apr 19 16:04:37 ip-172-31-4-32.eu-west-2.compute.internal systemd[1]: kmm.service: Control process exited, code=exited status=2
Apr 19 16:04:37 ip-172-31-4-32.eu-west-2.compute.internal systemd[1]: Failed to start Kubernetes Multi-master.
Apr 19 16:04:37 ip-172-31-4-32.eu-west-2.compute.internal systemd[1]: kmm.service: Unit entered failed state.
Apr 19 16:04:37 ip-172-31-4-32.eu-west-2.compute.internal systemd[1]: kmm.service: Failed with result 'exit-code'.
Apr 19 16:04:47 ip-172-31-4-32.eu-west-2.compute.internal systemd[1]: kmm.service: Service hold-off time over, scheduling restart.
Apr 19 16:04:47 ip-172-31-4-32.eu-west-2.compute.internal systemd[1]: Stopped Kubernetes Multi-master.
Apr 19 16:04:47 ip-172-31-4-32.eu-west-2.compute.internal systemd[1]: Starting Kubernetes Multi-master...
Apr 19 16:04:47 ip-172-31-4-32.eu-west-2.compute.internal docker[25216]: time="2017-04-19T16:04:47Z" level=info msg="Cloud Provider Initialized ["aws"]"
Apr 19 16:04:47 ip-172-31-4-32.eu-west-2.compute.internal docker[25216]: time="2017-04-19T16:04:47Z" level=info msg="Getting "kmm-asset-key" key value..."
Apr 19 16:04:47 ip-172-31-4-32.eu-west-2.compute.internal docker[25216]: time="2017-04-19T16:04:47Z" level=info msg="Assets not present in etcd...
Apr 19 16:04:47 ip-172-31-4-32.eu-west-2.compute.internal docker[25216]: "
Apr 19 16:04:47 ip-172-31-4-32.eu-west-2.compute.internal docker[25216]: time="2017-04-19T16:04:47Z" level=info msg="Transaction didn't succeed - we didn't create lock!"
Apr 19 16:04:47 ip-172-31-4-32.eu-west-2.compute.internal docker[25216]: time="2017-04-19T16:04:47Z" level=info msg="Lock allready created..."
Apr 19 16:04:47 ip-172-31-4-32.eu-west-2.compute.internal docker[25216]: time="2017-04-19T16:04:47Z" level=info msg="Getting "kmm-asset-lock" key value..."
Apr 19 16:04:47 ip-172-31-4-32.eu-west-2.compute.internal docker[25216]: time="2017-04-19T16:04:47Z" level=info msg="Lock (key - "kmm-asset-lock") not obtained, TTL exists:"2017-04-19T16:05:13Z""
Apr 19 16:05:07 ip-172-31-4-32.eu-west-2.compute.internal docker[25216]: time="2017-04-19T16:05:07Z" level=info msg="Getting "kmm-asset-key" key value..."
Apr 19 16:05:07 ip-172-31-4-32.eu-west-2.compute.internal docker[25216]: time="2017-04-19T16:05:07Z" level=info msg="Assets not present in etcd...
Apr 19 16:05:08 ip-172-31-4-32.eu-west-2.compute.internal docker[25216]: "
Apr 19 16:05:08 ip-172-31-4-32.eu-west-2.compute.internal docker[25216]: time="2017-04-19T16:05:08Z" level=info msg="Transaction didn't succeed - we didn't create lock!"
Apr 19 16:05:08 ip-172-31-4-32.eu-west-2.compute.internal docker[25216]: time="2017-04-19T16:05:08Z" level=info msg="Lock allready created..."
Apr 19 16:05:08 ip-172-31-4-32.eu-west-2.compute.internal docker[25216]: time="2017-04-19T16:05:08Z" level=info msg="Getting "kmm-asset-lock" key value..."
Apr 19 16:05:08 ip-172-31-4-32.eu-west-2.compute.internal docker[25216]: time="2017-04-19T16:05:08Z" level=info msg="Lock (key - "kmm-asset-lock") not obtained, TTL exists:"2017-04-19T16:05:13Z""
```

**Comment by lewismarshall** [comment]
The fix for this has to include tests:

# Write interfaces (for etcd and kubeadm) to test kmm.GetAssets().
# Tests for this use case (failed control plane, re-obtain lock).

**Comment by lewismarshall** [comment]
The bug could be fixed by #39 but needs a specific unit or integration test to ensure.
